homeassistant/components/azure_event_hub/hub.py

```diff
@@ -0,0 +1,178 @@
+"""Support for Azure Event Hubs."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable, Mapping
+from datetime import datetime
+import json
+import logging
+from typing import Any
+
+from azure.eventhub import EventData, EventDataBatch
+from azure.eventhub.exceptions import EventHubError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import MATCH_ALL
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.json import JSONEncoder
+from homeassistant.util.dt import utcnow
+
+from .client import AzureEventHubClient
+from .const import (
+    CONF_MAX_DELAY,
+    CONF_SEND_INTERVAL,
+    DATA_FILTER,
+    DOMAIN,
+    NON_SEND_STATES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AzureEventHub:
```

Comment:
Please do the hub refactor in a separate PR. Move it back to where it was and keep the signature intact.